### PR TITLE
DCOS-45813: Implement basic plans tab ("v0")

### PR DESCRIPTION
**Pull request by Poltergeist**
This introduces the plans tab to the detail page for sdk packages

Closes DCOS-45813
## Testing

1. Install a SDK package e.g. `Cassandra`
2. Visit the Detail page.
3. Visit the Plans tab

## Trade-offs

*N/A*

## Dependencies

This is depending on #3503 

## Screenshots

image

### CODE REVIEW

**Comment by GeorgiSTodorov** [comment]
I installed kafka, opened the plans tab and it was empty. The console contained some `console.log`s coming from `ServicePlansModel.ts`.
Same for cassandra.

**Comment by Poltergeist** [comment]
Unfortunately this has to do with https://github.com/dcos/dcos-ui/pull/3503#discussion_r247006487 I had a locally patched version allowing 200 and 202 that is why it worked on my machine. :-( you can still check this if you alter the mentioned line on this branch. I am going to rebase this branch as soon as this is fixed on the other Pr.

**Comment by GeorgiSTodorov** [comment]
@Poltergeist , okay. The `console.log`s were actually because I was using an older version of the branch. But, as you said, in the newest version it still doesn't work. I will check what you mentioned.
One more thing I noticed:
screenshot from 2019-01-11 10-06-58

**Comment by GeorgiSTodorov** [comment]
@Poltergeist , I changed the line you mentioned, but it is still not working. In what way did you change it to get it to work?

**Comment by Poltergeist** [comment]
@GeorgiSTodorov The any problem I already saw, but that should not block. 

My current diff looks like this:

```diff
diff --git a/plugins/services/src/js/data/index.ts b/plugins/services/src/js/data/index.ts
index 9808fa921..d1f7270d8 100644
--- a/plugins/services/src/js/data/index.ts
+++ b/plugins/services/src/js/data/index.ts
@@ -48,7 +48,7 @@ function isPlansQueryArgs(args: GeneralArgs): args is PlansQueryArgs {
 }
 
 function handleAPIError(reqResp: RequestResponse<any>, message: string = "") {
-  if (reqResp.code !== 200) {
+  if (![200, 202].includes(reqResp.code)) {
     const respMessage =
       reqResp.response && typeof reqResp.response === "object"
         ? JSON.stringify(reqResp.response)
@@ -84,7 +84,12 @@ export const resolvers = ({
               reqResp: RequestResponse<ServicePlanResponse>
             ): ServicePlanResponse[] => {
               handleAPIError(reqResp, "Service plan detail");
-              return [ServicePlanDetailResolver(reqResp.response)];
+              return [
+                ServicePlanDetailResolver({
+                  name: args.name,
+                  ...reqResp.response
+                })
+              ];
             }
           );
 
@@ -110,7 +115,10 @@ export const resolvers = ({
               return fetchServicePlanDetail(parent.name, plan.name).map(
                 (reqResp: RequestResponse<ServicePlanResponse>) => {
                   handleAPIError(reqResp, "Service plan detail");
-                  return ServicePlanDetailResolver(reqResp.response);
+                  return ServicePlanDetailResolver({
+                    name: plan.name,
+                    ...reqResp.response
+                  });
                 }
               );
             });
```

**Comment by GeorgiSTodorov** [comment]
Still doesn't work for me. I will wait for the rebase in order to test properly.

**Comment by Poltergeist** [comment]
Hmm should with the latest changes.

File: src/js/components/Page.d.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  actions?: [];
+  addButton?: Array<{}> | {};
+  breadCrumbs?: React.ReactNode;
+  supplementalContent?: React.ReactNode;
+  tabs?: [];
+  disabledActions?: boolean;
+}
+
+interface PageProps {
+  className?: [] | {} | string;
```
Comment (mperrotti) [inline src/js/components/Page.d.ts]: I see in the original propTypes from `src/js/components/Page.js` that `className` can be an object, but when does that actually happen?

Comment (Poltergeist) [reply src/js/components/Page.d.ts]: I have no idea I just copied the props basically. I will look it up later.

**Comment by Poltergeist** [comment]
@GeorgiSTodorov @mperrotti All your concerns should be solved now. I had to remove the Page component from the errormessage and the loading screen although I kept the typing changes inside of this PR, which is no totally unrelated. But the error shows now and also the loadingscreen shows now.

**Comment by GeorgiSTodorov** [comment]
Five snaphots tests are failing locally and system tests are failing here.
screenshot from 2019-01-16 09-13-38

**Comment by GeorgiSTodorov** [comment]
System tests are failing again.

**Pull request merged**